acquisitions/src/main/resources/thunderjet/mod-orders/features/independent-acquisitions-unit-for-ordering-and-receiving.feature

```diff
@@ -0,0 +1,720 @@
+        {
+          "productId": "9780764354113",
+          "productIdType": "8261054f-be78-422d-bd51-4ed9f33c3422",
+          "qualifier": "(paperback)"
+        }
+      ],
+      "acqUnitIds": ["#(acqUnitId2)"]
+    }
+    """
+    When method POST
+    Then status 201
+
+  ## Receive/Edit piece checks
+  Scenario: Verify Edit piece functionality
+    # Assign acqUnit1 to Order, open and receive this Order. Assign acqUnit2 to related Title.
+    # Prepare user that has only acqUnit1, try to edit related piece - operation should be forbidden
+    # because for any operation with piece we check acq unit from related Title, and user does not have acqUnit2 assigned.
+    # 1. Create a composite order
+    * configure headers = headersUser
+    Given path 'orders/composite-orders'
+    And request
+    """
+    {
+      id: '#(orderId2)',
+      vendor: '#(globalVendorId)',
+      orderType: 'One-Time',
+      acqUnitIds: ['#(acqUnitId1)']
+    }
+    """
+    When method POST
+    Then status 201
+
+    # 2. Create an order line
+    Given path 'orders/order-lines'
+    * def poLine = read('classpath:samples/mod-orders/orderLines/minimal-order-line.json')
+    * set poLine.id = poLineId2
+    * set poLine.purchaseOrderId = orderId2
+    * set poLine.fundDistribution[0].fundId = fundId
+    And request poLine
+    When method POST
+    Then status 201
+
+    # 3. Open the order
+    Given path 'orders/composite-orders', orderId2
+    When method GET
+    Then status 200
+
+    * def order = $
+    * set order.workflowStatus = 'Open'
+
+    Given path 'orders/composite-orders', orderId2
+    And request order
+    When method PUT
+    Then status 204
+
+    # 4. Receive the piece
+    # 4.1 Get the id of piece created when the order was opened
+    Given path 'orders/pieces'
+    And param query = 'poLineId==' + poLineId2
+    When method GET
+    Then status 200
+    And match $.totalRecords == 1
+    * def pieceId = $.pieces[0].id
+
+    # 4.2 Receive it
+    Given path 'orders/check-in'
+    And request
+    """
+    {
+      toBeCheckedIn: [
+        {
+          checkedIn: 1,
+          checkInPieces: [
+            {
+              id: "#(pieceId)",
+              itemStatus: "In process",
+              locationId: "#(globalLocationsId)"
+            }
+          ],
+          poLineId: "#(poLineId2)"
+        }
+      ],
+      totalRecords: 1
+    }
+    """
+    When method POST
+    Then status 200
+    And match $.receivingResults[0].processedSuccessfully == 1
+    * call pause 500
+
+    # 5. Create acq unit to assign to title
+    * configure headers = headersAdmin
+    Given path 'acquisitions-units/units'
+    And request
+    """
+    {
+      "id": '#(acqUnitId3)',
+      "protectUpdate": true,
+      "protectCreate": true,
+      "protectDelete": true,
+      "protectRead": true,
+      "name": "testAcqUnit3"
+    }
+    """
+    When method POST
+    Then status 201
+
+    # 6. Retrieve title and assign acqUnit2
```

Comment:
comment and code should match, in comment we have acqUnit2, in code - acqUnit3

Reply:
Fixed